src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Windows.cs

```diff
@@ -131,14 +132,53 @@ public NamedPipeTest_CurrentUserOnly_Windows(TestAccountImpersonator testAccount
             _testAccountImpersonator = testAccountImpersonator;
         }
 
+        [OuterLoop]
+        [ConditionalFact(nameof(IsAdminOnSupportedWindowsVersions))]
+        public async Task Connection_UnderDifferentUsers_CurrentUserOnlyOnServer_InvalidClientConnectionAttempts_DoNotBlockSuccessfulClient()
+        {
+            string name = PipeStreamConformanceTests.GetUniquePipeName();
+            ManualResetEvent @event = new ManualResetEvent(false);
+
+            Task invalidClient = Task.Run(() =>
+            {
+                // invalid non-current user tries to connect to server.
+                _testAccountImpersonator.RunImpersonated(() =>
+                {
+                    using var client = new NamedPipeClientStream(".", name, PipeDirection.In, PipeOptions.Asynchronous);
+                    Assert.Throws<UnauthorizedAccessException>(() =>
+                    {
+                        @event.WaitOne();
+                        client.Connect();
+                    });
+                });
+            });
+
+            // valid client tries to connect and succeeds.
+            using var client = new NamedPipeClientStream(".", name, PipeDirection.In, PipeOptions.Asynchronous);
+            Task validClient = Task.Run(() =>
+            {
+                @event.WaitOne();
+                client.Connect();
+            });
+
+            using var server = new NamedPipeServerStream(name, PipeDirection.InOut, 1, PipeTransmissionMode.Byte, PipeOptions.Asynchronous | PipeOptions.CurrentUserOnly);
+            @event.Set();
```

Comment:
There's a very high liklihood that both Connects will be initiated before the server begins waiting.  There's also a high liklihood that the valid client will connect before the invalid one does.

In the original test I'd proposed, the invalid client was looping, trying to connect over and over, and only once it started that loop would the valid client try to connect, the goal of the test being to ensure the invalid client couldn't starve out the valid client.  That's no longer happening here, so I'm not sure what this test is actually testing.  With the invalid client making a single connection attempt, eventually that attempt will end and the valid client can connect.  And if that were broken, that would be caught by just a simple test with no client/client concurrency that simply tried to connect from an invalid client and then sequentially tried to connect from a valid client.

Reply:
I can bring the original test back and keep this version as well.
fwiw: this was the old version that issues invalid connects in a loop:
https://github.com/dotnet/runtime/pull/72956/commits/c94d060f23cacce6b61d1588f4c9956ebba06f85#diff-23b3a1e5a562e7712c4adff12019bbc8485dd0566627dc77808efa97efee2b30R134

Reply:
I agree that this test is not worth keeping, because it will so rarely do anything useful. If you can tweak the test you brought back below, I think it will do it correctly, and you can delete this.